app/workers/extract_fulltext_job.rb

```diff
@@ -0,0 +1,80 @@
+#-- encoding: UTF-8
+#-- copyright
+# OpenProject is a project management system.
+# Copyright (C) 2012-2017 the OpenProject Foundation (OPF)
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2017 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See doc/COPYRIGHT.rdoc for more details.
+#++
+
+require 'text_extractor'
+
+class ExtractFulltextJob < ApplicationJob
+  # queue_as :text_extraction
+
+  def initialize(attachment_id)
+    @attachment_id = attachment_id
+  end
+
+  def perform
+    return unless attachment = find_attachment(@attachment_id)
+
+    text = TextExtractor::Resolver.new(attachment.diskfile, attachment.content_type).text if attachment.readable?
+
+    if OpenProject::Database.allows_tsv?
+      attachment_filename = attachment.file.file.filename
+      update_with_tsv(attachment.id, attachment_filename, text)
+    else
+      attachment.update(fulltext: text)
+    end
+  end
+
+  private
+
+  def update_with_tsv(id, filename, text)
+    language = OpenProject::Configuration.main_content_language
+    update_sql = <<-SQL
+          UPDATE "attachments" SET
+            fulltext = '%s',
+            "fulltext_tsv" = to_tsvector('%s', '%s'),
+            "file_tsv" = to_tsvector('%s', '%s')
+          WHERE ID = %s;
+    SQL
```

Comment:
```
Attachment
 .where(id: id)
 .update_all(['fulltext = ?, fulltext_tsv = to_tsvector(?, ?), file_tsv = to_tsvector(?, ?), 
             text, 
             language, 
             OpenProject::FullTextSearch.normalize_text(text), 
             language, 
             OpenProject::FullTextSearch.normalize_text(filename)])
``` 
should do the same while sanitizing at the same time.

Reply:
As written above, I believe transforming to ascii to be faulty and unnecessary. In addition, I could imagine it being slow when a file having 1 million chars is processed like this.